### PR TITLE
Expose `applications` and `exports` fields in `InventoryTarget` and Jsonnet `inventory()` callback

### PR DESCRIPTION
**Pull request by simu**
## Proposed Changes

* Restore the Kapitan <= 0.32 behavior of exposing the Reclass `applications`  (which Commodore depends on) and `exports` fields in the inventory

## Docs and Tests

* [x] Tests added
* [ ] Updated documentation

### CODE REVIEW

File: kapitan/resources.py [hunk 1]
```diff
@@ -278,7 +278,11 @@ def inventory(search_paths: list, target_name: str = None, inventory_path: str =
 
     if target_name:
         target = inv.get_target(target_name)
-        return {"parameters": target.parameters, "classes": target.classes}
+        return {
+            "parameters": target.parameters,
+            "classes": target.classes,
+            "applications": target.applications,
+        }
```
Comment (MatteoVoges) [inline kapitan/resources.py:285]: I think at this point we can simply return `dataclasses.asdict(target)`

**Review by MatteoVoges — APPROVED** [review]
Thank you for the fix. Do you know of any more special keys in reclass, that perhaps need to be exposed?

File: kapitan/inventory/inv_reclass.py [hunk 2]
```diff
@@ -44,6 +43,8 @@ def render_targets(self, targets: list = None, ignore_class_notfound: bool = Fal
             for target_name, rendered_target in rendered_inventory["nodes"].items():
                 self.targets[target_name].parameters = rendered_target["parameters"]
                 self.targets[target_name].classes = rendered_target["classes"]
+                self.targets[target_name].applications = rendered_target["applications"]
```
Comment (ademariag) [inline kapitan/inventory/inv_reclass.py:46]: Could we not also copy all key/values in `rendered_target`? why are we hardcoding specific keys?

Comment (ademariag) [reply kapitan/inventory/inv_reclass.py:46]: I think I see why, it is required for the keys to be known. ok

**Review by ademariag — APPROVED** [review]

File: kapitan/inventory/inv_reclass.py (hunk 2 above)
Comment (simu) [reply kapitan/inventory/inv_reclass.py:46]: I have no hard opinion here. I didn't find a helper for dataclasses which generates a dataclass from a dict (although I didn't search for very long), so I ended up just copying the fields.

**Pull request merged**